framework/Optimizers/SimulatedAnnealing.py

```diff
@@ -0,0 +1,717 @@
+      @ In, solutionExport, DataObject, optional, a PointSet to hold the solution
+      @ Out, None
+    """
+    RavenSampled.initialize(self, externalSeeding=externalSeeding, solutionExport=solutionExport)
+    self.info = {}
+    for var in self.toBeSampled:
+      self.info['amp_'+var] = None
+      self.info['delta_'+var] = None
+    # queue up the first run for each trajectory
+    for traj, init in enumerate(self._initialValues):
+      self._submitRun(init,traj,self.getIteration(traj))
+
+  def initializeTrajectory(self, traj=None):
+    """
+      Handles the generation of a trajectory.
+      @ In, traj, int, optional, label to use
+      @ Out, traj, int, new trajectory number
+    """
+    traj = RavenSampled.initializeTrajectory(self)
+    self._acceptHistory[traj] = deque(maxlen=self._maxHistLen)
+    self._acceptRerun[traj] = False
+    self._convergenceInfo[traj] = {'persistence': 0}
+    for criteria in self._convergenceCriteria:
+      self._convergenceInfo[traj][criteria] = False
+    return traj
+
+  def _submitRun(self, point, traj, step, moreInfo=None):
+    """
+      Submits a single run with associated info to the submission queue
+      @ In, point, dict, point to submit
+      @ In, traj, int, trajectory identifier
+      @ In, step, int, iteration number identifier
+      @ In, moreInfo, dict, optional, additional run-identifying information to track
+      @ Out, None
+    """
+    info = {}
+    if moreInfo is not None:
+      info.update(moreInfo)
+    info.update({'traj': traj,
+                  'step': step
+                })
+    # NOTE: explicit constraints have been checked before this!
+    self.raiseADebug('Adding run to queue: {} | {}'.format(point, info))
+    self._submissionQueue.append((point, info))
+  # END queuing Runs
+  # * * * * * * * * * * * * * * * *
+
+  ###############
+  # Run Methods #
+  ###############
+  def _useRealization(self, info, rlz):
+    """
+      Used to feedback the collected runs into actionable items within the sampler.
+      @ In, info, dict, identifying information about the realization
+      @ In, rlz, dict, realized realization
+      @ In, optVal, float, value of objective variable (corrected for min/max)
+      @ Out, None
+    """
+    traj = info['traj']
+    info['optVal'] = rlz[self._objectiveVar]
+    self.incrementIteration(traj)
+    self._resolveNewOptPoint(traj, rlz, rlz[self._objectiveVar], info)
+    if self._stepTracker[traj]['opt'] == None:
+      # revert to the last accepted point
+      rlz = self._optPointHistory[traj][-1][0]
+      info = self._optPointHistory[traj][-1][1]
+      info['step'] = self.getIteration(traj)
+      optVal = rlz[self._objectiveVar]
+    iter = int(self.getIteration(traj) +1) # Is that ok or should we always keep the traj in case I have multiple trajectories in parallel?
+    fraction = iter/self.limit
+    currentPoint = self._collectOptPoint(rlz)
+    self.T0 = self._temperature(fraction)
```

Comment:
why do we need to store the self.T0? it is used only in the method below.

Reply:
removed.